paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -77,6 +78,7 @@ public class PaparazziPlugin : Plugin<Project> {
   private fun <T> setupPaparazzi(project: Project, variants: DomainObjectSet<T>) where T : BaseVariant, T : TestedVariant {
     project.addTestDependency()
     val nativePlatformFileCollection = project.setupNativePlatformDependency()
+    val snapshotOutputDir = project.layout.projectDirectory.dir("src/test/snapshots")
```

Comment:
I think this needs to read from the system property

`System.getProperty("paparazzi.snapshot.dir")`

Reply:
Good call, didn't catch that. Should that have been this property too then?

Reply:
Oh I see. So then this clean task would need to depend on that task so the property is setup for use

Reply:
the definition of `snapshotOutputDir` isn't variant-specific, so should be safe to define outside of `variants.all{}`

Reply:
while it isn't today, it probably should be.  for example, see: https://github.com/cashapp/paparazzi/issues/595.

I propose moving this back into the variants.all block and making the `delete` task variant-aware as well (looks like `cleanRecordVariants` is fine as is).

Reply:
i agree that it should be, but ATM if the `delete` task is made variant-aware, it will delete the same directory multiple times.
It would likely take some more sophisticated task graph wiring to ensure *all* `delete` tasks are run before *any* `record` task in case someone calls the global `cleanRecord`